src/RemoteFileAPI/Remote.ts

```diff
@@ -24,24 +24,40 @@ export class Remote {
     this.connection?.close(eventCodeWhenIntentionallyStoppingConnection);
   }
 
-  public startConnection(): void {
+  public startConnection(autoConnectAttempt = 0): void {
     const address = (Settings.UseWssForRemoteFileApi ? "wss" : "ws") + "://" + this.ipaddr + ":" + this.port;
+
+    //This tracks if a connection was established to prevent redundant toasts
+    let successfullyConnected = false;
+
     try {
       this.connection = new WebSocket(address);
     } catch (error) {
       console.error(error);
       showErrorMessage(address, String(error));
       return;
     }
-    this.connection.addEventListener("error", (e: Event) => showErrorMessage(address, JSON.stringify(e)));
+
+    //log connection errors on manual and the first auto connect attempts
+    if (autoConnectAttempt <= 1)
+      this.connection.addEventListener("error", (e: Event) => showErrorMessage(address, JSON.stringify(e)));
+
```

Comment:
```suggestion
    this.connection.addEventListener("error", (e: Event) => {
      if (autoConnectAttempt <= 1 || successfullyConnected) {
        showErrorMessage(address, JSON.stringify(e));
      }
    });
```
The flow of setting this event listener in two places is a bit awkward, and more importantly I'm concerned about possible edge-condition bugs due to the different timing and possibility of this.connection referring to a different object inside the callback. Unconditionally setting the listener and then doing the checks inside makes everything easier to validate.

Reply:
yeah, makes sense. This seems really obvious in hindsight